type_helpers.go

```diff
@@ -117,3 +128,70 @@ func SMTPAuthValue(v SMTPAuthType) *SMTPAuthType {
 func String(v string) *string {
 	return &v
 }
+
+// Unsettable is a wrapper that can be used for marshaling attributes which use
+// significant nil values, but still require `omitempty` behavior.
+//
+// This is generally useful for PATCH requests, where attributes with zero
+// values are intentionally not marshaled into the request payload.
+//
+// Helper functions are provided to help avoid pointer wrangling.
+//
+// NOTE: This type is only for use with outbound requests. It will cause errors
+// if used to unmarshal API responses, since the jsonapi module does not yet
+// support custom unmarshaling.
+
+type Unsettable[T any] struct {
+	Value *T
+}
+
+func UnsettableString(v string) *Unsettable[string] {
+	return &Unsettable[string]{&v}
+}
+
+func NullString() *Unsettable[string] {
+	return &Unsettable[string]{}
+}
+
+func UnsettableInt(v int) *Unsettable[int] {
+	return &Unsettable[int]{&v}
+}
+
+func NullInt() *Unsettable[string] {
+	return &Unsettable[string]{}
+}
+
+func UnsettableBool(v bool) *Unsettable[bool] {
+	return &Unsettable[bool]{&v}
+}
+
+func NullBool() *Unsettable[bool] {
+	return &Unsettable[bool]{}
+}
+
+var iso8601TimeFormat = "2006-01-02T15:04:05Z"
+
+func UnsettableTime(v time.Time) *Unsettable[time.Time] {
+	return &Unsettable[time.Time]{&v}
+}
+
+func NullTime() *Unsettable[time.Time] {
+	return &Unsettable[time.Time]{}
+}
+
+func (t *Unsettable[T]) MarshalJSON() ([]byte, error) {
+	if t == nil || t.Value == nil {
+		return json.RawMessage(`null`), nil
+	}
+
+	var b []byte
+	var err error
+
+	val := reflect.ValueOf(t.Value)
+	if val.Type().Kind() == reflect.Ptr && val.Elem().Type() == reflect.TypeOf(time.Time{}) {
+		return json.Marshal(val.Elem().Interface().(time.Time).Format(iso8601TimeFormat))
```

Comment:
should this be this instead?
```suggestion
		return json.Marshal(val.Elem().Interface().(time.Time).Format(time.RFC3339))
```